Bug: test_drop_counters start to fail due to PR https://github.com/sonic-net/sonic-mgmt/pull/22257

### Is it platform specific

generic

### Importance or Severity

Critical

### Description of the bug

After the PR https://github.com/sonic-net/sonic-mgmt/pull/22257 merged, the tests/drop_packets/test_drop_counters.py start to fail due to a same test issue:

```
AttributeError: '' object has no attribute 'facts'
do_test = .do_counters_test at 0x7f346d0a4f70>
ptfadapter = 
duthosts = []
enum_rand_one_per_hwsku_frontend_hostname = 'arc-switch1025'
setup = {'dut_to_ptf_port_map': {'Ethernet0': 0, 'Ethernet100': 25, 'Ethernet104': 26, 'Ethernet108': 27, ...}, 'intf_per_name..., '10.9.0.0/16', '10.212.0.0/16', '10.245.0.0/16', ...], ...}, 'neighbor_sniff_ports': [16, 0, 1, 17, 18, 2, ...], ...}
fanouthost = { os: 'onyx', hostname: 'arc-switch1035', device_type: 'FanoutLeaf' }
pkt_fields = {'ipv4_dst': '10.0.0.1', 'ipv4_src': '1.1.1.1', 'ipv6_dst': 'FC00::2', 'ipv6_src': 'ffff::101:101', 'tcp_sport': 1234, 'tcp_dport': 4321}
ports_info = {'asic_index': None, 'dst_mac': '50:6b:4b:8f:ce:40', 'dut_iface': 'Ethernet20', 'ptf_tx_port_id': 5, ...}

    def test_reserved_dmac_drop(do_test, ptfadapter, duthosts, enum_rand_one_per_hwsku_frontend_hostname,
                                setup, fanouthost, pkt_fields, ports_info):  # noqa: F811
        """
        @summary: Verify that packet with reserved DMAC is dropped and L2 drop counter incremented
        @used_mac_address:
            01:80:C2:00:00:05 - reserved for future standardization
            01:80:C2:00:00:08 - provider Bridge group address
        """
        duthost = duthosts[enum_rand_one_per_hwsku_frontend_hostname]
        if not fanouthost:
            pytest.skip("Test case requires explicit fanout support")
    
        # Marvell ASIC specific ACL rule injection
>       if fanouthost.facts["asic_type"] == "marvell-teralynx":

do_test    = .do_counters_test at 0x7f346d0a4f70>
duthost    = 
duthosts   = []
enum_rand_one_per_hwsku_frontend_hostname = 'arc-switch1025'
fanouthost = { os: 'onyx', hostname: 'arc-switch1035', device_type: 'FanoutLeaf' }
pkt_fields = {'ipv4_dst': '10.0.0.1', 'ipv4_src': '1.1.1.1', 'ipv6_dst': 'FC00::2', 'ipv6_src': 'ffff::101:101', 'tcp_sport': 1234, 'tcp_dport': 4321}
ports_info = {'asic_index': None, 'dst_mac': '50:6b:4b:8f:ce:40', 'dut_iface': 'Ethernet20', 'ptf_tx_port_id': 5, ...}
ptfadapter = 
setup      = {'dut_to_ptf_port_map': {'Ethernet0': 0, 'Ethernet100': 25, 'Ethernet104': 26, 'Ethernet108': 27, ...}, 'intf_per_name..., '10.9.0.0/16', '10.212.0.0/16', '10.245.0.0/16', ...], ...}, 'neighbor_sniff_ports': [16, 0, 1, 17, 18, 2, ...], ...}

drop_packets/test_drop_counters.py:364: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
common/devices/fanout.py:50: in __getattr__
    return getattr(self.host, module_name)
        module_name = 'facts'
        self       = { os: 'onyx', hostname: 'arc-switch1035', device_type: 'FanoutLeaf' }
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = , module_name = 'facts'

    def __getattr__(self, module_name):
        if self.host.has_module(module_name):
            self.module_name = module_name
            self.module = getattr(self.host, module_name)
    
            return self._run
>       raise AttributeError(
            "'%s' object has no attribute '%s'" % (self.__class__, module_name)
            )
E       AttributeError: '' object has no attribute 'facts'

module_name = 'facts'
self       = 

common/devices/base.py:62: AttributeError
```

### Steps to Reproduce

Run the test script tests/drop_packets/test_drop_counters.py

### Actual Behavior and Expected Behavior

4 cases failure due to the same issue listed above:
test_equal_smac_dmac_drop
test_multicast_smac_drop
test_non_routable_igmp_pkts
test_reserved_dmac_drop


### Relevant log output

```shell
AttributeError: '' object has no attribute 'facts'
do_test = .do_counters_test at 0x7f346d0a4f70>
ptfadapter = 
duthosts = []
enum_rand_one_per_hwsku_frontend_hostname = 'arc-switch1025'
setup = {'dut_to_ptf_port_map': {'Ethernet0': 0, 'Ethernet100': 25, 'Ethernet104': 26, 'Ethernet108': 27, ...}, 'intf_per_name..., '10.9.0.0/16', '10.212.0.0/16', '10.245.0.0/16', ...], ...}, 'neighbor_sniff_ports': [16, 0, 1, 17, 18, 2, ...], ...}
fanouthost = { os: 'onyx', hostname: 'arc-switch1035', device_type: 'FanoutLeaf' }
pkt_fields = {'ipv4_dst': '10.0.0.1', 'ipv4_src': '1.1.1.1', 'ipv6_dst': 'FC00::2', 'ipv6_src': 'ffff::101:101', 'tcp_sport': 1234, 'tcp_dport': 4321}
ports_info = {'asic_index': None, 'dst_mac': '50:6b:4b:8f:ce:40', 'dut_iface': 'Ethernet20', 'ptf_tx_port_id': 5, ...}

    def test_reserved_dmac_drop(do_test, ptfadapter, duthosts, enum_rand_one_per_hwsku_frontend_hostname,
                                setup, fanouthost, pkt_fields, ports_info):  # noqa: F811
        """
        @summary: Verify that packet with reserved DMAC is dropped and L2 drop counter incremented
        @used_mac_address:
            01:80:C2:00:00:05 - reserved for future standardization
            01:80:C2:00:00:08 - provider Bridge group address
        """
        duthost = duthosts[enum_rand_one_per_hwsku_frontend_hostname]
        if not fanouthost:
            pytest.skip("Test case requires explicit fanout support")
    
        # Marvell ASIC specific ACL rule injection
>       if fanouthost.facts["asic_type"] == "marvell-teralynx":

do_test    = .do_counters_test at 0x7f346d0a4f70>
duthost    = 
duthosts   = []
enum_rand_one_per_hwsku_frontend_hostname = 'arc-switch1025'
fanouthost = { os: 'onyx', hostname: 'arc-switch1035', device_type: 'FanoutLeaf' }
pkt_fields = {'ipv4_dst': '10.0.0.1', 'ipv4_src': '1.1.1.1', 'ipv6_dst': 'FC00::2', 'ipv6_src': 'ffff::101:101', 'tcp_sport': 1234, 'tcp_dport': 4321}
ports_info = {'asic_index': None, 'dst_mac': '50:6b:4b:8f:ce:40', 'dut_iface': 'Ethernet20', 'ptf_tx_port_id': 5, ...}
ptfadapter = 
setup      = {'dut_to_ptf_port_map': {'Ethernet0': 0, 'Ethernet100': 25, 'Ethernet104': 26, 'Ethernet108': 27, ...}, 'intf_per_name..., '10.9.0.0/16', '10.212.0.0/16', '10.245.0.0/16', ...], ...}, 'neighbor_sniff_ports': [16, 0, 1, 17, 18, 2, ...], ...}

drop_packets/test_drop_counters.py:364: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
common/devices/fanout.py:50: in __getattr__
    return getattr(self.host, module_name)
        module_name = 'facts'
        self       = { os: 'onyx', hostname: 'arc-switch1035', device_type: 'FanoutLeaf' }
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = , module_name = 'facts'

    def __getattr__(self, module_name):
        if self.host.has_module(module_name):
            self.module_name = module_name
            self.module = getattr(self.host, module_name)
    
            return self._run
>       raise AttributeError(
            "'%s' object has no attribute '%s'" % (self.__class__, module_name)
            )
E       AttributeError: '' object has no attribute 'facts'

module_name = 'facts'
self       = 

common/devices/base.py:62: AttributeError
```

### Output of `show version`

```shell
202511
```

### Attach files (if any)

_No response_

--- a/tests/drop_packets/drop_packets.py
+++ b/tests/drop_packets/drop_packets.py
@@ -104,7 +104,7 @@ def fanouthost(duthosts, enum_rand_one_per_hwsku_frontend_hostname, fanouthosts,
             fanout.restore_drop_counter_config()
 
     if fanout:
-        if fanout.facts["asic_type"] == "marvell-teralynx":
+        if hasattr(fanout, "facts") and fanout.facts.get("asic_type") == "marvell-teralynx":
             # Check and clean up existing REDIRECT_VLAN ACL table if present.
             check_output = fanout.shell("show acl table", module_ignore_errors=True)
             if "REDIRECT_VLAN" in check_output["stdout"]:
@@ -663,7 +663,7 @@ def test_equal_smac_dmac_drop(do_test, ptfadapter, setup, fanouthost,
     src_mac = ports_info["dst_mac"]
 
     # Marvell ASIC specific ACL rule injection
-    if fanouthost.facts["asic_type"] == "marvell-teralynx":
+    if hasattr(fanouthost, "facts") and fanouthost.facts.get("asic_type") == "marvell-teralynx":
         drop_counter_config(fanouthost)
 
     if fanouthost.os == 'onyx':
@@ -712,7 +712,7 @@ def test_multicast_smac_drop(do_test, ptfadapter, setup, fanouthost,
                    pkt_fields["ipv4_dst"], pkt_fields["ipv4_src"])
 
     # Marvell ASIC specific ACL rule injection
-    if fanouthost.facts["asic_type"] == "marvell-teralynx":
+    if hasattr(fanouthost, "facts") and fanouthost.facts.get("asic_type") == "marvell-teralynx":
         drop_counter_config(fanouthost)
 
     if fanouthost.os == 'onyx':

--- a/tests/drop_packets/test_drop_counters.py
+++ b/tests/drop_packets/test_drop_counters.py
@@ -361,7 +361,7 @@ def test_reserved_dmac_drop(do_test, ptfadapter, duthosts, enum_rand_one_per_hws
         pytest.skip("Test case requires explicit fanout support")
 
     # Marvell ASIC specific ACL rule injection
-    if fanouthost.facts["asic_type"] == "marvell-teralynx":
+    if hasattr(fanouthost, "facts") and fanouthost.facts.get("asic_type") == "marvell-teralynx":
         drop_counter_config(fanouthost)
 
     reserved_mac_addr = ["01:80:C2:00:00:05", "01:80:C2:00:00:08"]

